service/src/main/java/org/ehrbase/service/ValidationServiceImp.java

```diff
@@ -131,6 +135,44 @@ public void check(Composition composition) throws Exception {
         check(composition.getArchetypeDetails().getTemplateId().getValue(), composition);
     }
 
+    @Override
+    public void check(EhrStatus ehrStatus) {
+
+        //case of a system generated ehr
+        if (ehrStatus == null)
+            return;
+
+        //first, check the built EhrStatus using the general Archie RM-Validator
+        List<RMObjectValidationMessage> rmObjectValidationMessages = RM_OBJECT_VALIDATOR.validate(ehrStatus);
+
+        if (!rmObjectValidationMessages.isEmpty()){
+            StringBuilder stringBuilder = new StringBuilder();
+            for (RMObjectValidationMessage rmObjectValidationMessage: rmObjectValidationMessages){
+                stringBuilder.append(rmObjectValidationMessage.toString());
+                stringBuilder.append("\n");
+            }
+            throw new ValidationException(stringBuilder.toString());
```

Comment:
I changed the exceptions to our own one. This is our general approach we decided on once. Would you agree here, or had you some custom use case in mind? If not, would you change the exception type in the other methods here, too?